plugins/services/src/js/types/ServicePlan.ts

```diff
@@ -9,6 +13,58 @@ export interface ServicePlan {
   strategy: string;
 }
 
+export interface ServicePlanElement {
+  type: "phase" | "step";
+  id: string;
+  name: string;
+  status: ServicePlanStatus;
+  strategy?: string;
+  phaseId?: string;
+  message?: string;
+  steps?: ServicePlanElement[];
+}
+
+export function flattenServicePlan(plan: ServicePlan): ServicePlanElement[] {
```

Comment:
`types/ServicePlan.ts` suggests that the file is for types only, shall we find another place for the `flattenServicePlan` and `compare` functions?

Reply:
is the best alternative to put these functions in a file in the `utils` directory?